File: java/src/test/java/javasearch/SearcherTest.java
package javasearch;

import org.junit.Test;

import java.io.File;

import static org.junit.Assert.*;

public class SearcherTest {

    public SearcherTest() {}

    private static SearchSettings getSettings() {
        SearchSettings settings = new SearchSettings();
        settings.setStartPath(".");
        settings.addSearchPattern("Searcher");
        return settings;
    }


    /*************************************************************
     * isSearchDir tests
     *************************************************************/
    @Test
    public void testisSearchDir_SingleDot_True() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        assertTrue(searcher.isSearchDir(new File(".")));
    }

    @Test
    public void testisSearchDir_DoubleDot_True() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        assertTrue(searcher.isSearchDir(new File("..")));
    }

    @Test
    public void testisSearchDir_IsHidden_False() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        assertFalse(searcher.isSearchDir(new File(".git")));
    }

    @Test
    public void testisSearchDir_IsHiddenIncludeHidden_True() {
        SearchSettings settings = getSettings();
        settings.setExcludeHidden(false);
        Searcher searcher = new Searcher(settings);
        assertTrue(searcher.isSearchDir(new File(".git")));
    }

    @Test
    public void testisSearchDir_NoPatterns_True() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        assertTrue(searcher.isSearchDir(new File("/Users")));
    }

    @Test
    public void testisSearchDir_MatchesInPattern_True() {
        SearchSettings settings = getSettings();
        settings.addInDirPattern("Search");
        Searcher searcher = new Searcher(settings);
        assertTrue(searcher.isSearchDir(new File("CsSearch")));
    }

    @Test
    public void testisSearchDir_MatchesOutPattern_False() {
        SearchSettings settings = getSettings();
        settings.addOutDirPattern("Search");
        Searcher searcher = new Searcher(settings);
        assertFalse(searcher.isSearchDir(new File("CsSearch")));
    }

    @Test
    public void testisSearchDir_DoesNotMatchInPattern_False() {
        SearchSettings settings = getSettings();
        settings.addInDirPattern("SearchFiles");
        Searcher searcher = new Searcher(settings);
        assertFalse(searcher.isSearchDir(new File("CsSearch")));
    }

    @Test
    public void testisSearchDir_DoesNotMatchOutPattern_True() {
        SearchSettings settings = getSettings();
        settings.addOutDirPattern("SearchFiles");
        Searcher searcher = new Searcher(settings);
        File dir = new File("CsSearch");
        assertTrue(searcher.isSearchDir(dir));
    }


    /*************************************************************
     * isSearchFile tests
     *************************************************************/

    @Test
    public void testIsSearchFile_NoExtensionsNoPatterns_True() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertTrue(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_MatchesInExtension_True() {
        SearchSettings settings = getSettings();
        settings.addInExtension("cs");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertTrue(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_DoesNotMatchInExtension_False() {
        SearchSettings settings = getSettings();
        settings.addInExtension("java");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertFalse(searcher.isSearchFile(file));
    }


    @Test
    public void testIsSearchFile_MatchesOutExtension_False() {
        SearchSettings settings = getSettings();
        settings.addOutExtension("cs");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertFalse(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_DoesNotMatchOutExtension_True() {
        SearchSettings settings = getSettings();
        settings.addOutExtension("java");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertTrue(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_MatchesInPattern_True() {
        SearchSettings settings = getSettings();
        settings.addInFilePattern("Search");
        Searcher searcher = new Searcher(settings);
        File file = new File("Searcher.cs");
        assertTrue(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_DoesNotMatchInPattern_False() {
        SearchSettings settings = getSettings();
        settings.addInFilePattern("Search");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertFalse(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_MatchesOutPattern_False() {
        SearchSettings settings = getSettings();
        settings.addOutFilePattern("Search");
        Searcher searcher = new Searcher(settings);
        File file = new File("Searcher.cs");
        assertFalse(searcher.isSearchFile(file));
    }

    @Test
    public void testIsSearchFile_DoesNotMatchOutPattern_True() {
        SearchSettings settings = getSettings();
        settings.addOutFilePattern("Search");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertTrue(searcher.isSearchFile(file));
    }


    /*************************************************************
     * IsArchiveSearchFile tests
     *************************************************************/

    @Test
    public void testIsArchiveSearchFile_NoExtensionsNoPatterns_True() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_MatchesInExtension_True() {
        SearchSettings settings = getSettings();
        settings.addInArchiveExtension("zip");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_DoesNotMatchInExtension_False() {
        SearchSettings settings = getSettings();
        settings.addInArchiveExtension("gz");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.isArchiveSearchFile(file));
    }


    @Test
    public void testIsArchiveSearchFile_MatchesOutExtension_False() {
        SearchSettings settings = getSettings();
        settings.addOutArchiveExtension("zip");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_DoesNotMatchOutExtension_True() {
        SearchSettings settings = getSettings();
        settings.addOutArchiveExtension("gz");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_MatchesInPattern_True() {
        SearchSettings settings = getSettings();
        settings.addInArchiveFilePattern("arch");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_DoesNotMatchInPattern_False() {
        SearchSettings settings = getSettings();
        settings.addInArchiveFilePattern("archives");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_MatchesOutPattern_False() {
        SearchSettings settings = getSettings();
        settings.addOutArchiveFilePattern("arch");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.isArchiveSearchFile(file));
    }

    @Test
    public void testIsArchiveSearchFile_DoesNotMatchOutPattern_True() {
        SearchSettings settings = getSettings();
        settings.addOutArchiveFilePattern("archives");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.isArchiveSearchFile(file));
    }

    /*************************************************************
     * FilterFile tests
     *************************************************************/

    @Test
    public void testFilterFile_IsHidden_False() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        File file = new File(".gitignore");
        assertFalse(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_IsHiddenIncludeHidden_True() {
        SearchSettings settings = getSettings();
        settings.setExcludeHidden(false);
        Searcher searcher = new Searcher(settings);
        File file = new File(".gitignore");
        assertTrue(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_ArchiveNoSearchArchives_False() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_ArchiveSearchArchives_True() {
        SearchSettings settings = getSettings();
        settings.setSearchArchives(true);
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_IsArchiveSearchFile_True() {
        SearchSettings settings = getSettings();
        settings.setSearchArchives(true);
        settings.addInArchiveExtension("zip");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertTrue(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_NotIsArchiveSearchFile_False() {
        SearchSettings settings = getSettings();
        settings.addOutExtension("zip");
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_ArchiveFileArchivesOnly_True() {
        SearchSettings settings = getSettings();
        settings.setArchivesOnly(true);
        Searcher searcher = new Searcher(settings);
        File file = new File("archive.zip");
        assertFalse(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_NoExtensionsNoPatterns_True() {
        SearchSettings settings = getSettings();
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertTrue(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_IsSearchFile_True() {
        SearchSettings settings = getSettings();
        settings.addInExtension("cs");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertTrue(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_NotIsSearchFile_False() {
        SearchSettings settings = getSettings();
        settings.addOutExtension("cs");
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertFalse(searcher.filterFile(file));
    }

    @Test
    public void testFilterFile_NonArchiveFileArchivesOnly_False() {
        SearchSettings settings = getSettings();
        settings.setArchivesOnly(true);
        Searcher searcher = new Searcher(settings);
        File file = new File("FileUtil.cs");
        assertFalse(searcher.filterFile(file));
    }
}

File: node/tests/searcher_test.js
/*
 * searcher_test.js
 *
 * Some nodeunit tests of searcher.js
 */

//var FileType = require('../nodesearch/filetype.js').FileType;
var FileUtil = require('../nodesearch/fileutil.js').FileUtil;
var Searcher = require('../nodesearch/searcher.js').Searcher;
var SearchSettings = require('../nodesearch/searchsettings.js').SearchSettings;

var testFile = FileUtil.expandPath("~/src/git/xsearch/shared/testFiles/testFile2.txt");

var getSettings = function() {
    var settings = new SearchSettings();
    settings.startPath = '.';
    settings.addSearchPattern ('Searcher');
    return settings;
    test.done()
}

/*************************************************************
 * isSearchDir tests
 *************************************************************/
exports.testisSearchDir_SingleDot_True = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    test.ok(searcher.isSearchDir("."));
    test.done()
}

exports.testisSearchDir_DoubleDot_True = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    test.ok(searcher.isSearchDir(".."));
    test.done()
}

exports.testisSearchDir_IsHidden_False = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    test.ok(!searcher.isSearchDir(".git"));
    test.done()
}

exports.testisSearchDir_IsHiddenIncludeHidden_True = function(test) {
    var settings = getSettings();
    settings.excludeHidden = false;
    var searcher = new Searcher(settings);
    test.ok(searcher.isSearchDir(".git"));
    test.done()
}

exports.testisSearchDir_NoPatterns_True = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    test.ok(searcher.isSearchDir("/Users"));
    test.done()
}

exports.testisSearchDir_MatchesInPattern_True = function(test) {
    var settings = getSettings();
    settings.addInDirPattern("Search");
    var searcher = new Searcher(settings);
    test.ok(searcher.isSearchDir("CsSearch"));
    test.done()
}

exports.testisSearchDir_MatchesOutPattern_False = function(test) {
    var settings = getSettings();
    settings.addOutDirPattern("Search");
    var searcher = new Searcher(settings);
    test.ok(!searcher.isSearchDir("CsSearch"));
    test.done()
}

exports.testisSearchDir_DoesNotMatchInPattern_False = function(test) {
    var settings = getSettings();
    settings.addInDirPattern("SearchFiles");
    var searcher = new Searcher(settings);
    test.ok(!searcher.isSearchDir("CsSearch"));
    test.done()
}

exports.testisSearchDir_DoesNotMatchOutPattern_True = function(test) {
    var settings = getSettings();
    settings.addOutDirPattern("SearchFiles");
    var searcher = new Searcher(settings);
    var dir = "CsSearch";
    test.ok(searcher.isSearchDir(dir));
    test.done()
}

/*************************************************************
 * isSearchFile tests
 *************************************************************/
exports.testIsSearchFile_NoExtensionsNoPatterns_True = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_MatchesInExtension_True = function(test) {
    var settings = getSettings();
    settings.addInExtension("cs");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_DoesNotMatchInExtension_False = function(test) {
    var settings = getSettings();
    settings.addInExtension("java");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(!searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_MatchesOutExtension_False = function(test) {
    var settings = getSettings();
    settings.addOutExtension("cs");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(!searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_DoesNotMatchOutExtension_True = function(test) {
    var settings = getSettings();
    settings.addOutExtension("java");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_MatchesInPattern_True = function(test) {
    var settings = getSettings();
    settings.addInFilePattern("Search");
    var searcher = new Searcher(settings);
    var file = "Searcher.cs";
    test.ok(searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_DoesNotMatchInPattern_False = function(test) {
    var settings = getSettings();
    settings.addInFilePattern("Search");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(!searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_MatchesOutPattern_False = function(test) {
    var settings = getSettings();
    settings.addOutFilePattern("Search");
    var searcher = new Searcher(settings);
    var file = "Searcher.cs";
    test.ok(!searcher.isSearchFile(file));
    test.done()
}

exports.testIsSearchFile_DoesNotMatchOutPattern_True = function(test) {
    var settings = getSettings();
    settings.addOutFilePattern("Search");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(searcher.isSearchFile(file));
    test.done()
}

/*************************************************************
 * IsArchiveSearchFile tests
 *************************************************************/
exports.testIsArchiveSearchFile_NoExtensionsNoPatterns_True = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_MatchesInExtension_True = function(test) {
    var settings = getSettings();
    settings.addInArchiveExtension("zip");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_DoesNotMatchInExtension_False = function(test) {
    var settings = getSettings();
    settings.addInArchiveExtension("gz");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_MatchesOutExtension_False = function(test) {
    var settings = getSettings();
    settings.addOutArchiveExtension("zip");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_DoesNotMatchOutExtension_True = function(test) {
    var settings = getSettings();
    settings.addOutArchiveExtension("gz");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_MatchesInPattern_True = function(test) {
    var settings = getSettings();
    settings.addInArchiveFilePattern("arch");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_DoesNotMatchInPattern_False = function(test) {
    var settings = getSettings();
    settings.addInArchiveFilePattern("archives");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_MatchesOutPattern_False = function(test) {
    var settings = getSettings();
    settings.addOutArchiveFilePattern("arch");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.isArchiveSearchFile(file));
    test.done()
}

exports.testIsArchiveSearchFile_DoesNotMatchOutPattern_True = function(test) {
    var settings = getSettings();
    settings.addOutArchiveFilePattern("archives");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.isArchiveSearchFile(file));
    test.done()
}

/*************************************************************
 * filterFile tests
 *************************************************************/
exports.testFilterFile_IsHidden_False = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var file = ".gitignore";
    test.ok(!searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_IsHiddenIncludeHidden_True = function(test) {
    var settings = getSettings();
    settings.excludeHidden = false;
    var searcher = new Searcher(settings);
    var file = ".gitignore";
    test.ok(searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_ArchiveNoSearchArchives_False = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_ArchiveSearchArchives_True = function(test) {
    var settings = getSettings();
    settings.searchArchives = true;
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_IsArchiveSearchFile_True = function(test) {
    var settings = getSettings();
    settings.searchArchives = true;
    settings.addInArchiveExtension("zip");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_NotIsArchiveSearchFile_False = function(test) {
    var settings = getSettings();
    settings.addOutExtension("zip");
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_ArchiveFileArchivesOnly_True = function(test) {
    var settings = getSettings();
    settings.archivesOnly = true;
    var searcher = new Searcher(settings);
    var file = "archive.zip";
    test.ok(!searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_NoExtensionsNoPatterns_True = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_IsSearchFile_True = function(test) {
    var settings = getSettings();
    settings.addInExtension("cs");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_NotIsSearchFile_False = function(test) {
    var settings = getSettings();
    settings.addOutExtension("cs");
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(!searcher.filterFile(file));
    test.done()
}

exports.testFilterFile_NonArchiveFileArchivesOnly_False = function(test) {
    var settings = getSettings();
    settings.archivesOnly = true;
    var searcher = new Searcher(settings);
    var file = "FileUtil.cs";
    test.ok(!searcher.filterFile(file));
    test.done()
}

/*************************************************************
 * searchLines test
 *************************************************************/
exports.TestSearchLines = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var lines = FileUtil.getFileLines(testFile);

    var results = searcher.searchLines(lines);
    test.ok(results.length == 2);

    var firstResult = results[0];
    var expectedFirstLineNum = 23;
    test.ok(firstResult.linenum == expectedFirstLineNum);
    var expectedFirstMatchStartIndex = 3;
    test.ok(firstResult.matchStartIndex === expectedFirstMatchStartIndex);
    var expectedFirstMatchEndIndex = 11;
    test.ok(firstResult.matchEndIndex === expectedFirstMatchEndIndex);

    var secondResult = results[1];
    var expectedSecondLineNum = 29;
    test.ok(secondResult.linenum === expectedSecondLineNum);
    var expectedSecondMatchStartIndex = 24;
    test.ok(secondResult.matchStartIndex === expectedSecondMatchStartIndex);
    var expectedSecondMatchEndIndex = 32;
    test.ok(secondResult.matchEndIndex === expectedSecondMatchEndIndex);

    test.done()
}

/*************************************************************
 * searchMultiLineString test
 *************************************************************/
exports.TestSearchMultiLineString = function(test) {
    var settings = getSettings();
    var searcher = new Searcher(settings);
    var contents = FileUtil.getFileContents(testFile);

    var results = searcher.searchMultiLineString(contents);
    test.ok(results.length == 2);

    var firstResult = results[0];
    var expectedFirstLineNum = 23;
    test.ok(firstResult.linenum == expectedFirstLineNum);
    var expectedFirstMatchStartIndex = 3;
    test.ok(firstResult.matchStartIndex === expectedFirstMatchStartIndex);
    var expectedFirstMatchEndIndex = 11;
    test.ok(firstResult.matchEndIndex === expectedFirstMatchEndIndex);

    var secondResult = results[1];
    var expectedSecondLineNum = 29;
    test.ok(secondResult.linenum === expectedSecondLineNum);
    var expectedSecondMatchStartIndex = 24;
    test.ok(secondResult.matchStartIndex === expectedSecondMatchStartIndex);
    var expectedSecondMatchEndIndex = 32;
    test.ok(secondResult.matchEndIndex === expectedSecondMatchEndIndex);

    test.done()
}
